Examples/Tests/plasma_lens/analysis.py

```diff
@@ -28,31 +28,34 @@
 ad = ds.all_data()
 
 # Get final position of the particles
-# The sorting by id is to get them in the correct order
 # There are two particles, one moves in x, the other in y.
-r_id = ad['electrons', 'particle_id'].v
-
-xx_sim = ad['electrons', 'particle_position_x'].v[np.argsort(r_id)][0]
-yy_sim = ad['electrons', 'particle_position_y'].v[np.argsort(r_id)][1]
-zz_sim0 = ad['electrons', 'particle_position_z'].v[np.argsort(r_id)][0]
-zz_sim1 = ad['electrons', 'particle_position_z'].v[np.argsort(r_id)][1]
-
-ux_sim = ad['electrons', 'particle_momentum_x'].v[np.argsort(r_id)][0]/m_e
-uy_sim = ad['electrons', 'particle_momentum_y'].v[np.argsort(r_id)][1]/m_e
-
-
-def applylens(x0, vx0, vz0, lens_length, lens_strength):
-    """Given the initial position at the start of the lens,
-    calculate the end position at the exit.
-    This assumes harmonic motion."""
-    w = np.sqrt(e/m_e*lens_strength)
-    A = np.sqrt(x0**2 + (vx0/w)**2)
-    phi = np.arctan2(-vx0, w*x0)
-    t = lens_length/vz0
-    x1 = A*np.cos(w*t + phi)
-    vx1 = -w*A*np.sin(w*t + phi)
+# The particles may not be in order, so determine which is which.
+i0 = np.argmax(np.abs(ad['electrons', 'particle_position_x'].v))
+i1 = np.argmax(np.abs(ad['electrons', 'particle_position_y'].v))
+
+xx_sim = ad['electrons', 'particle_position_x'].v[i0]
+yy_sim = ad['electrons', 'particle_position_y'].v[i1]
```

Comment:
I'm not familiar with this test and I'm not sure I understand: why do we set here (and use later on) the x position only for the particle index `i0` and the y position only for the particle index `i1`? Similar question for the momenta `ux_sim` and `uy_sim` below.

Reply:
I've updated the comments, so hopefully it.s more clear.